### ISSUE TITLE
be able to handle header file changes

### ISSUE THREAD

**Issue by jpoimboe** [dynup/kpatch#315]
Making a trivial change to a header file causes a lot of problems for kpatch-build.  For example, modifying a comment in include/linux/ptrace.h results in what looks like the entire kernel tree getting rebuilt, causing at least two problems with kpatch-build (and probably more).

Problem 1: it dies with "a struct definition change was detected" because kpatch-build detected that arch/x86/kernel/asm-offsets.s changed.

Problem 2: If I remove that error check from kpatch-build then it fails with 

```
/root/.kpatch/src/scripts/Makefile.build:44: /root/.kpatch/src/arch/x86/kvm/../../../virt/kvm/Makefile: No such file or directory
make[2]: *** No rule to make target `/root/.kpatch/src/arch/x86/kvm/../../../virt/kvm/Makefile'.  Stop.
make[1]: *** [arch/x86/kvm/../../../virt/kvm/kvm_main.o] Error 2
```

Probably because the makefile for virt/kvm/kvm_main.c is in arch/x86/kvm.

So basically, we might want to be able to handle all the special cases involved in the entire kernel tree getting rebuilt, even if there were only a few real changes.
